sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -131,18 +145,98 @@ protected void beforeTest() {
     protected void afterTest() {
     }
 
+    /**
+     * Get offset which determines the rolling strategy.
+     *
+     * @return The offset according to day of the week and platform information
+     */
+    private static int getOffset() {
+        LocalDate today = LocalDate.now();
+        buildCalendarMap();
+        buildPlatformList();
+        return calendarMap.get(today.getDayOfWeek()) + getPlatFormOffset();
+    }
+
+    private static Integer getPlatFormOffset() {
+        String currentOs = System.getProperty("os.name").toLowerCase(Locale.ROOT);;
+        String currentJdk = System.getProperty("java.version").toLowerCase(Locale.ROOT);;
+        return platformList.stream().filter(platform ->
+            currentOs.trim().toLowerCase(Locale.ROOT).contains(platform.split(",")[0].toLowerCase(Locale.ROOT))
+            && currentJdk.trim().toLowerCase(Locale.ROOT).startsWith(platform.split(",")[1]
+                .toLowerCase(Locale.ROOT))
+        ).map(platformList::indexOf).findFirst().orElse(null);
+    }
+
+    private static void buildPlatformList() {
+        if (platformList != null) {
+            return;
+        }
+        platformList = new ArrayList<>();
+        platformList.add("win,8");
+        platformList.add("win,11");
+        platformList.add("mac,8");
+        platformList.add("mac,11");
+        platformList.add("linux,8");
+        platformList.add("linux,11");
+    }
+
+    private static void buildCalendarMap() {
+        if (calendarMap != null) {
+            return;
+        }
+        calendarMap = new HashMap<>();
+        calendarMap.put(DayOfWeek.MONDAY, 0);
+        calendarMap.put(DayOfWeek.TUESDAY, 1);
+        calendarMap.put(DayOfWeek.WEDNESDAY, 2);
+        calendarMap.put(DayOfWeek.THURSDAY, 3);
+        calendarMap.put(DayOfWeek.FRIDAY, 4);
+        calendarMap.put(DayOfWeek.SATURDAY, 5);
+        calendarMap.put(DayOfWeek.SUNDAY, 6);
+    }
```

Comment:
This map is not required. `DayOfWeek.MONDAY.getValue()` will return the int value of that day.